samcli/lib/build/app_builder.py

```diff
@@ -366,17 +367,11 @@ def _stream_lambda_image_build_logs(self, build_logs: List[Dict[str, str]], func
         function_name str
             Name of the function that is being built
         """
-        for log in build_logs:
-            if log:
-                log_stream = log.get("stream")
-                error_stream = log.get("error")
-
-                if error_stream:
-                    raise DockerBuildFailed(f"{function_name} failed to build: {error_stream}")
-
-                if log_stream:
-                    self._stream_writer.write(str.encode(log_stream))
-                    self._stream_writer.flush()
+        build_log_streamer = LogStreamer(self._stream_writer)
+        try:
+            build_log_streamer.stream_progress(build_logs)
+        except LogStreamError as ex:
+            raise DockerBuildFailed(f"{function_name} failed to build: {str(ex)}")
```

Comment:
nit: `raise DockerBuildFailed(..) from ex` for [exception chaining](https://docs.python.org/3/tutorial/errors.html#exception-chaining).